packages/frontend/src/features/notification/stores/useNotificationStore.tsx

```diff
@@ -0,0 +1,69 @@
+import { create } from 'zustand'
+import { NotificationAction } from '../types/NotificationTypes'
+
+interface Notification {
+    id: number
+    type: string
+    message: string
+    time: string
+    isRead: boolean
+    actions?: NotificationAction[]
+    targetId?: string
+}
+
+interface NotificationStore {
+    notifications: Notification[]
+    showNotificationDot: boolean
+    addNotification: (notification: Notification) => void
+    markAsRead: (id: number) => void
+    loadNotifications: () => void
+    reset: () => void
+    clearNotificationDot: () => void
+}
+
+// Define initial state
+const initialState = {
+    notifications: [] as Notification[],
+    showNotificationDot: false,
+}
+
+export const useNotificationStore = create<NotificationStore>((set) => ({
+    ...initialState,
+    addNotification: (notification) =>
+        set((state) => {
+            const updatedNotifications = [...state.notifications, notification]
+            localStorage.setItem(
```

Comment:
You can use zustand integration to persist store data. 
[Reference](https://zustand.docs.pmnd.rs/integrations/persisting-store-data)

Reply:
Integrated to using `persist`.